radix-engine/src/blueprints/epoch_manager/validator.rs

```diff
@@ -763,6 +881,19 @@ impl ValidatorBlueprint {
 
         Ok(())
     }
+
+    /// Returns an amount of stake units to be minted when [`xrd_amount`] of XRDs is being staked.
+    fn calculate_stake_unit_amount(
+        xrd_amount: Decimal,
+        total_stake_xrd_amount: Decimal,
+        total_stake_unit_supply: Decimal,
+    ) -> Decimal {
+        if total_stake_xrd_amount.is_zero() {
+            xrd_amount
+        } else {
+            xrd_amount * total_stake_unit_supply / total_stake_xrd_amount
+        }
```

Comment:
I always ask for extra caution for math expressions. Is it possible to reach a state where `total_stake_unit_supply == 0 && total_stake_xrd_amount != 0`? If so, what's the behaviour?